.eslintrc.js

```diff
@@ -24,6 +24,8 @@ module.exports = {
     'function-paren-newline': 'off',
     'function-call-argument-newline': 'off',
     'default-param-last': 'off',
+    'no-redeclare': 'off',
```

Comment:
Are those actually neccesary? If possible, replace them with eslint-ignore comments (or even better, fix the issue without any changes to the linting rules)

Reply:
I disabled that rule to replace it with the TypeScript version of the same rule. The [ESLint documentation](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-redeclare.md) says "note you must disable the base rule as it can report incorrect errors".

Specifically, the base `no-redeclare` rule reports these errors:

```
...\fnbr.js\src\structures\STWMeleeWeaponSchematic.ts
  9:7  error  'STWMeleeWeaponSchematic' is already defined  no-redeclare

...\fnbr.js\src\structures\STWRangedWeaponSchematic.ts
  9:7  error  'STWRangedWeaponSchematic' is already defined  no-redeclare

...\fnbr.js\src\structures\STWTrapSchematic.ts
  10:7  error  'STWTrapSchematic' is already defined  no-redeclare

D:\working\github\fnbr.js\src\structures\STWWeaponSchematic.ts
  10:7  error  'STWWeaponSchematic' is already defined  no-redeclare

✖ 4 problems (4 errors, 0 warnings)
```

Those files define an interface and a class with the same name; the interface constrains the types of fields that were introduced in the base class. It's a valid pattern in TypeScript, and I don't know of a better way to write it, but ESLint's base rules are unaware of TypeScript declaration merging.

Reply:
I think it's not necessary to use an interface. Take a look at <https://github.com/fnbrjs/fnbr.js/blob/master/src/structures/SentFriendMessage.ts>.
In my opinion that's also a less complex way than defining an interface that provides the overwritten types